test/providers/cuda_helpers.cpp

```diff
@@ -202,23 +216,40 @@ int cuda_fill(CUcontext context, CUdevice device, void *ptr, size_t size,
         return -1;
     }
 
+    // set required context
+    CUcontext curr_context = nullptr;
+    set_context(context, &curr_context);
```

Comment:
nit: Do we need to set a context? I though that `cuMemsetD32` automatically detects the context of `dst_ptr` and `src_ptr`.

Reply:
test doesn't work without this (only on windows) - cuGetErrorString() returns "invalid argument"